dart/lib/src/utils/http_sanitizer.dart

```diff
@@ -0,0 +1,82 @@
+import '../protocol.dart';
+import 'url_details.dart';
+
+class HttpSanitizer {
+  static final RegExp _authRegExp = RegExp("(.+://)(.*@)(.*)");
+
+  /// Parse and sanitize url data for sentry.io
+  static UrlDetails? sanitizeUrl(String? url) {
+    if (url == null) {
+      return null;
+    }
+
+    final queryIndex = url.indexOf('?');
+    final fragmentIndex = url.indexOf('#');
+
+    if (queryIndex > -1 && fragmentIndex > -1 && fragmentIndex < queryIndex) {
+      // url considered malformed because of fragment position
+      return UrlDetails();
+    } else {
+      final uri = Uri.parse(url);
+      final urlWithAuthRemoved = _urlWithAuthRemoved(uri._url());
+      return UrlDetails(
+          url: urlWithAuthRemoved.isEmpty ? null : urlWithAuthRemoved,
+          query: uri.query.isEmpty ? null : uri.query,
+          fragment: uri.fragment.isEmpty ? null : uri.fragment);
+    }
+  }
+
+  static Map<String, String>? sanitizedHeaders(Map<String, String>? headers) {
+    var mutableHeaders =
+        headers != null ? Map<String, String>.from(headers) : null;
+    mutableHeaders?.remove('authorization');
+    mutableHeaders?.remove('Authorization');
+    mutableHeaders?.remove('cookies');
+    mutableHeaders?.remove('Cookies');
+    mutableHeaders?.remove('cookie');
+    mutableHeaders?.remove('Cookie');
```

Comment:
Is that a data structure that is case-insensitive?

Reply:
Solved with the addititions